From 5ef5cfc0f66ef01d18070237d8e64c8031d3b29f Mon Sep 17 00:00:00 2001
From: AnilKumar Ch <anilkumar@ti.com>
Date: Mon, 2 Jan 2012 14:59:20 +0545
Subject: [PATCH 0747/1092] am335x: regulator: tps65217 fix for BeagleBone A3
 and greater

This patch fixes the problem that tps65217B regulator support
is applicable only for BeagleBone A3 and greater. And tps65217B
regulator is not applicable for the BeagleBone's whose revision
verion is < A3

Signed-off-by: AnilKumar Ch <anilkumar@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c | 31 +++++++++++++++++++++++++++----
 1 file changed, 27 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index f414ca2..e9e2c28 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -1487,6 +1487,32 @@ static void mmc0_init(int evm_id, int profile)
 	return;
 }
 
+static struct i2c_board_info tps65217_i2c_boardinfo[] = {
+	{
+		I2C_BOARD_INFO("tps65217", TPS65217_I2C_ID),
+		.platform_data  = &beaglebone_tps65217_info,
+	},
+};
+
+static void tps65217_init(int evm_id, int profile)
+{
+	struct i2c_adapter *adapter;
+	struct i2c_client *client;
+
+	/* I2C1 adapter request */
+	adapter = i2c_get_adapter(1);
+	if (!adapter) {
+		pr_err("failed to get adapter i2c1\n");
+		return;
+	}
+
+	client = i2c_new_device(adapter, tps65217_i2c_boardinfo);
+	if (!client)
+		pr_err("failed to register tps65217 to i2c1\n");
+
+	i2c_put_adapter(adapter);
+}
+
 static void mmc0_no_cd_init(int evm_id, int profile)
 {
 	setup_pin_mux(mmc0_no_cd_pin_mux);
@@ -1671,6 +1697,7 @@ static struct evm_dev_cfg beaglebone_old_dev_cfg[] = {
 
 /* Beaglebone Rev A3 and after */
 static struct evm_dev_cfg beaglebone_dev_cfg[] = {
+	{tps65217_init,	DEV_ON_BASEBOARD, PROFILE_NONE},
 	{mii1_init,	DEV_ON_BASEBOARD, PROFILE_NONE},
 	{usb0_init,	DEV_ON_BASEBOARD, PROFILE_NONE},
 	{usb1_init,	DEV_ON_BASEBOARD, PROFILE_NONE},
@@ -1969,10 +1996,6 @@ static struct i2c_board_info __initdata am335x_i2c_boardinfo[] = {
 		I2C_BOARD_INFO("tps65910", TPS65910_I2C_ID1),
 		.platform_data  = &am335x_tps65910_info,
 	},
-	{
-		I2C_BOARD_INFO("tps65217", TPS65217_I2C_ID),
-		.platform_data  = &beaglebone_tps65217_info,
-	},
 };
 
 static struct omap_musb_board_data musb_board_data = {
-- 
1.7.11.2

